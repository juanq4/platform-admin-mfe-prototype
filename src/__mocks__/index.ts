export * from "./contexts";
export * from "./data";
export * from "./utils";
