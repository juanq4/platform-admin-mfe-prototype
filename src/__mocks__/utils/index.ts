export * from "./helpers";
export * from "./permissions";
export * from "./wrappers";
