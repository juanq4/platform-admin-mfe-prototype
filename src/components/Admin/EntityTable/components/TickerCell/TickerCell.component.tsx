import { isEmpty } from "@q4/nimbus-ui";
import React, { memo } from "react";
import { TickerCellIdModel } from "./TickerCell.definition";
import type { TickerCellProps } from "./TickerCell.definition";

const TickerCellBase = (props: TickerCellProps): JSX.Element => {
  const { id, value } = props;

  const idModel = new TickerCellIdModel(id);
  const ticker = isEmpty(value) ? "" : value[0].split(".")[0];

  return <span id={idModel.id}>{ticker}</span>;
};

export const TickerCell = memo(TickerCellBase);
