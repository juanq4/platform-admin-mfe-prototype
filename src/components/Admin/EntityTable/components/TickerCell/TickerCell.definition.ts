import { AutomationListId, IdModelBase } from "@q4/nimbus-ui";
import type { Key } from "react";
import type { Organization } from "../../../../../definitions";
import type { EntityTableCellPropsBase } from "../../EntityTable.definition";

export type TickerCellProps = EntityTableCellPropsBase<Organization["identifiers"]>;

export class TickerCellListId extends AutomationListId<TickerCellIdModel> {
  constructor(parentId: string, prefix = "TickerCell-", postfix?: string) {
    super(parentId, prefix, TickerCellIdModel, postfix);
  }
}

export class TickerCellIdModel extends IdModelBase {
  constructor(id: IdModelBase["id"], index?: Key, postfix?: string) {
    super(id, index, postfix);
  }
}
