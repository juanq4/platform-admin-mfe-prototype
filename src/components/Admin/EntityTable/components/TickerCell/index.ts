export * from "./TickerCell.component";
export * from "./TickerCell.definition";
export * from "./TickerCell.utils";
