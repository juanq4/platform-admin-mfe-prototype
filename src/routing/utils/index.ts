export * from "./api";
export * from "./s3checks";
export * from "./versionManager";
