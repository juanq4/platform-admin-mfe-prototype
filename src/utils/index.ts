export * from "./account";
export * from "./api";
export * from "./entitlement";
export * from "./error";
export * from "./feature";
export * from "./logout";
export * from "./organization";
export * from "./permission";
export * from "./role";
export * from "./route";
export * from "./subscription";
export * from "./user";
export * from "./environment";
