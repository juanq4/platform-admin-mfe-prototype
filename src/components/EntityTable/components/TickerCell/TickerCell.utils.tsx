import { TickerCell } from "./TickerCell.component";
import type { TickerCellListId, TickerCellProps } from "./TickerCell.definition";

export function mapIdToTickerCell(params: TickerCellProps, listId: TickerCellListId): JSX.Element {
  const id = listId?.getId(params?.data?.id);

  return <TickerCell {...params} id={id} />;
}
