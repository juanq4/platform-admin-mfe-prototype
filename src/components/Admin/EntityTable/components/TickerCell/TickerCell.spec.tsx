import React from "react";
import { CheckMarkCellIdModel } from "..";
import { getAppWrapper } from "../../../../../__mocks__";
import { render, screen } from "../../../../../utils/testUtils";
import { TickerCell } from "./TickerCell.component";
import { TickerCellListId } from "./TickerCell.definition";
import { mapIdToTickerCell } from "./TickerCell.utils";

const mockTickerCellValues = [null, [], ["ART.NYE"]];

function testTickerCell(id: string): void {
  const element = screen.queryByTestId(id);
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
}

describe("TickerCell Component", () => {
  const idModel = new CheckMarkCellIdModel("MockCheckMarkCellId");

  test("7386712: [Given] no props are provided [Expect] the component should render", () => {
    render(<TickerCell value={null} />, { container: getAppWrapper() });
    const rootElement = screen.queryByTestId("root");
    expect(rootElement).toMatchSnapshot();
  });

  test("7434031: [Given] value is 'null' [Expect] the ticker to be 'empty'", () => {
    render(<TickerCell id={idModel.id} value={mockTickerCellValues[0]} />);
    testTickerCell(idModel.id);
  });

  test("7434032: [Given] value is '' [Expect] the ticker to be 'empty'", () => {
    render(<TickerCell id={idModel.id} value={mockTickerCellValues[1]} />);
    testTickerCell(idModel.id);
  });

  test("7434033: [Given] value is 'ART.NYE' [Expect] the ticker to be 'ART'", () => {
    render(<TickerCell id={idModel.id} value={mockTickerCellValues[2]} />);
    testTickerCell(idModel.id);
  });
});

describe("mapIdToTickerCell", () => {
  const listId = new TickerCellListId("MockTickerCell");
  const mockData = { id: "4dec165f-4093-4182-bd72-88754" };

  test("7386713: [Given] the function is passed not no props [Expect] the component should render", () => {
    render(mapIdToTickerCell(null, null), { container: getAppWrapper() });
    const rootElement = screen.queryByTestId("root");
    expect(rootElement).toMatchSnapshot();
  });

  test("7434034: [Given] value is 'null' [Expect] the ticker to be 'empty'", () => {
    render(mapIdToTickerCell({ data: { ...mockData }, value: mockTickerCellValues[0] }, listId));
    const id = listId.getId(mockData.id);
    testTickerCell(id);
  });

  test("7434035: [Given] value is '' [Expect] the ticker to be 'empty'", () => {
    render(mapIdToTickerCell({ data: { ...mockData }, value: mockTickerCellValues[1] }, listId));
    const id = listId.getId(mockData.id);
    testTickerCell(id);
  });

  test("7434036: [Given] value is 'ART.NYE' [Expect] the ticker to be 'ART'", () => {
    render(mapIdToTickerCell({ data: { ...mockData }, value: mockTickerCellValues[2] }, listId));
    const id = listId.getId(mockData.id);
    testTickerCell(id);
  });
});
