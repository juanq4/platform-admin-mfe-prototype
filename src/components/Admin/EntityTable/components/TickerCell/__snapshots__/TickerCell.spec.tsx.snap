// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TickerCell Component 7386712: [Given] no props are provided [Expect] the component should render 1`] = `
<div
  id="root"
>
  <span />
</div>
`;

exports[`TickerCell Component 7434031: [Given] value is 'null' [Expect] the ticker to be 'empty' 1`] = `
<span
  id="MockCheckMarkCellId"
/>
`;

exports[`TickerCell Component 7434032: [Given] value is '' [Expect] the ticker to be 'empty' 1`] = `
<span
  id="MockCheckMarkCellId"
/>
`;

exports[`TickerCell Component 7434033: [Given] value is 'ART.NYE' [Expect] the ticker to be 'ART' 1`] = `
<span
  id="MockCheckMarkCellId"
>
  ART
</span>
`;

exports[`mapIdToTickerCell 7386713: [Given] the function is passed not no props [Expect] the component should render 1`] = `
<div
  id="root"
>
  <span />
</div>
`;

exports[`mapIdToTickerCell 7434034: [Given] value is 'null' [Expect] the ticker to be 'empty' 1`] = `
<span
  id="MockTickerCellTickerCell-4dec165f-4093-4182-bd72-88754"
/>
`;

exports[`mapIdToTickerCell 7434035: [Given] value is '' [Expect] the ticker to be 'empty' 1`] = `
<span
  id="MockTickerCellTickerCell-4dec165f-4093-4182-bd72-88754"
/>
`;

exports[`mapIdToTickerCell 7434036: [Given] value is 'ART.NYE' [Expect] the ticker to be 'ART' 1`] = `
<span
  id="MockTickerCellTickerCell-4dec165f-4093-4182-bd72-88754"
>
  ART
</span>
`;
